ops/ccs-ops-misc/load_test/v1/regression_suite.py

```diff
@@ -6,173 +6,140 @@
-      """Returns the next beneficiary ID in this BFDUser's list of IDs.
-
-      This method pops (that is, takes the topmost item) the next beneficiary
-      ID in this instance's list of beneficiary IDs and returns it. If no
-      more IDs are available, this method will stop the test run. 
-      """
-      
-      if len(self.bene_ids) == 0:
-            errors.no_data_stop_test(self)
-
-      return self.bene_ids.pop()
-
-    def get_mbi(self) -> int:
-      """Returns the next MBI in this BFDUser's list of MBIs.
-
-      This method pops (that is, takes the topmost item) the next MBI 
-      in this instance's list of MBIs and returns it. If no
-      more MBIs are available, this method will stop the test run. 
-      """
-      
-      if len(self.mbis) == 0:
-            errors.no_data_stop_test(self)
-
-      return self.mbis.pop()
-
-    def get_cursor_path(self) -> str:
-      """Returns the next cursor path in this BFDUser's list of URL paths.
-
-      This method pops (that is, takes the topmost item) the next contract cursor 
-      URL path in this instance's list of contract paths and returns it. If no
-      more paths are available, this method will stop the test run. 
-      """
-
-      if len(self.cursor_list) == 0:
-        errors.no_data_stop_test(self)
-
-      return self.cursor_list.pop()
-
-    def get(self, base_path: str, params: Dict[str, str] = {}, headers: Dict[str, str] = {}, name: str = ''):
-      """Sends a GET request to the endpoint at base_path with the various query string parameters and headers specified.
-
-      This method extends Locust's HttpUser::client.get() method to make creating the requests 
-      nicer. Specifically, the query string parameters are specified as a separate dictionary
-      opposed to part of the path, the cert and verify arguments (which will never change)
-      are already set, and Cache-Control headers are automatically set to ensure caching is
-      disabled.
-      """
-
-      self.client.get(create_url_path(base_path, params),
-                      cert=client_cert,
-                      verify=server_public_key,
-                      headers={**headers, 'Cache-Control': 'no-store, no-cache'},
-                      name=name)
+from common.validation import SLA_V1_BASELINE
+from locust import task
+
+class BFDUser(BFDUserBase):
+    DATA_REQUIRED = [ 'BENE_IDS', 'MBIS', 'CONTRACT_IDS' ]
+    SLA_BASELINE = SLA_V1_BASELINE
+
 
     @task
     def coverage_test_id_count(self):
-        self.get(f'/v1/fhir/Coverage', params={'beneficiary': f'{self.get_bene_id()}', '_count':'10'},
-                name='/v1/fhir/Coverage search by id / count=10')
+        def make_url(instance):
+            return create_url_path('/v1/fhir/Coverage', {
+                'beneficiary': instance.bene_ids.pop(),
+                '_count': '10'
+            })
+
+        self.run_task(name='/v1/fhir/Coverage search by id / count=10', url_callback=make_url)
```

Comment:
I notice that this pattern of defining a `make_url` function which returns the string created by `create_url_path` is almost universal across all of the test methods. I'm wondering if we could reduce a lot of this boilerplate (since it's so ubiquitous) by adding an overload for `run_task` that does this and takes the parameters that `get` normally would, i.e.:

```python
# Name can be whatever you want
def run_url_path_task(self, path: str, query_params: Dict[str, str] = {}, headers: Dict[str, str] = {}, name: str):
  def make_url(instance):
     ...
  self.run_task(name=name, url_callback=make_url)
```
Then, this would become something more like:

```python
def coverage_test_id_count(self):
  self.run_url_path_task('/v1/fhir/Coverage', {
                'beneficiary': instance.bene_ids.pop(),
                '_count': '10'
            }, name='/v1/fhir/Coverage search by id / count=10')
```